### PR TITLE
internal: add CI checks for major Linux distros and fix build errors

### PR DESCRIPTION
**Pull request by nic11**
> [!IMPORTANT]
> Enhanced CI workflow with multi-distro support, improved build scripts, and minor code fixes.
> 
>   - **CI Workflow**:
>     - Renamed `.github/workflows/pr-ubuntu-docker.yml` to `pr-linux-variants.yml`.
>     - Added support for multiple Linux distributions in `pr-linux-variants.yml` including Ubuntu 22.04, 24.04, 24.10, Debian 12, and Arch Linux.
>     - Added scripts for setting up dependencies for each supported Linux distribution in `misc/deps_linux/`.
>   - **Build Script**:
>     - Updated `build.sh` to dynamically set compilers based on availability.
>     - Added `--clean-vcpkg` and `--print-env` options to `build.sh`.
>   - **Code Improvements**:
>     - Added missing includes for `<algorithm>` and `<cstdint>` in several files including `WRLD.cpp`, `OpcodesImplementation.cpp`, `ActivateEvent.h`, `CustomEvent.h`, `EatItemEvent.h`, and `ReadBookEvent.h`.
>     - Fixed return statement in `SimdJsonInputArchive::Serialize` in `SimdJsonInputArchive.h` to satisfy GCC requirements.
>     - Improved error messages in `yarn.cmake` for better debugging.
> 
> <sup>This description was created by </sup>[Ellipsis](https://www.ellipsis.dev?ref=skyrim-multiplayer%2Fskymp&utm_source=github&utm_medium=referral)<sup> for f747f7bb989c4d7b7b4b9891203aee7412dc7edc. It will automatically update as commits are pushed.</sup>

### CODE REVIEW

File: serialization/include/archives/SimdJsonInputArchive.h [hunk 1]
```diff
@@ -65,6 +65,7 @@ class SimdJsonInputArchive
   SimdJsonInputArchive& Serialize(T& output)
   {
     static_assert(!sizeof(T), "can only parse to std::string");
+    return *this; // gcc wants it
```
Comment (Pospelove) [inline serialization/include/archives/SimdJsonInputArchive.h:68]: I compiled with gcc without this line

Comment (nic11) [reply serialization/include/archives/SimdJsonInputArchive.h:68]: It gives warning

Comment (nic11) [reply serialization/include/archives/SimdJsonInputArchive.h:68]: Hm actually
```
/home/roma/my/skymp/skymp-arch/serialization/include/archives/SimdJsonInputArchive.h: In member function ‘SimdJsonInputArchive& SimdJsonInputArchive::Serialize(T&)’:
/home/roma/my/skymp/skymp-arch/serialization/include/archives/SimdJsonInputArchive.h:69:3: error: no return statement in function returning non-void [-Werror=return-type]
   69 |   }
      |   ^
cc1plus: some warnings being treated as errors
```
It's a warning that's treated as error

Comment (nic11) [reply serialization/include/archives/SimdJsonInputArchive.h:68]:
```
ubuntu@9b94c15f0dde:/home/roma/my/skymp/skymp-arch$ c++ --version
c++ (Ubuntu 13.3.0-6ubuntu2~24.04) 13.3.0
Copyright (C) 2023 Free Software Foundation, Inc.
This is free software; see the source for copying conditions.  There is NO
warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
```

File: Dockerfile [hunk 2]
```diff
@@ -21,10 +21,7 @@ RUN useradd -m skymp
 # It contains everything that should be installed on the system.
 FROM skymp-runtime-base AS skymp-build-base
 
-# TODO: replace with a call to ubuntu2204?
```
Comment (nic11) [inline Dockerfile]: Deliberately skipped Dockerfile for now to resolve in a separate PR

**Comment by nic11** [comment]
@Pospelove should the title be
* fix: add CI checks for major Linux distros and fix build errors
* internal: add CI checks for major Linux distros and fix build errors

?

**Comment by nic11** [comment]
@ellipsis-dev update description

**Pull request merged**